core/src/main/scala/org/apache/spark/scheduler/TaskLocation.scala

```diff
@@ -35,16 +37,26 @@ case class ExecutorCacheTaskLocation(override val host: String, executorId: Stri
 
 /**
  * A location on a host.
+ * The host names will be resolved to IP address in case SPARK_LOCAL_HOSTNAME is unset
```

Comment:
You may want to refer "SPARK-10149" in the comments so that others can have more clues to understand the changes here....